pkg/archive/archive.go

```diff
@@ -55,9 +55,10 @@ type Entry interface {
 	Path() string                                              // Absolute path to the entry in the archive.
 	Length() uint64                                            // Uncompressed data length.
 	CompressedLength() uint64                                  // Compressed data length.
+	CompressedAs(compressionMethod uint16) bool                // Whether the entry is compressed using the given method.
```

Comment:
Isn't the `8` value for deflate only relevant in the context of a ZIP? Theoretically we could use `Archive` for a CBR/RAR file as well.

To remove any ambiguity, what do you think of adding an explicit enum with `Deflate` case in the toolkit instead of using ints?

Having `CompressionMethod() CompressionMethod` might be more convenient too.

Reply:
Technically, the latest version of the [ZIP spec](https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT) supports other compression formats as well. I think the correct term would be that the EPUB standard only supports modes 0 and 8. I think it's worth keeping the door open for other compression modes for other container formats that we might support, or whose specs may change to support other methods in the future.

I've added your suggested `CompressionMethod` enum